data-prepper-plugins/rds-source/src/main/java/org/opensearch/dataprepper/plugins/source/rds/datatype/postgres/handler/NumericTypeHandler.java

```diff
@@ -30,15 +37,62 @@ private Object parseNumericValue(PostgresDataType columnType, String textValue)
             case BIGSERIAL:
                 return Long.parseLong(textValue);
             case REAL:
-                return Float.parseFloat(textValue);
+                return parseRealValue(textValue);
             case DOUBLE_PRECISION:
-                return Double.parseDouble(textValue);
+                return parseDoublePrecisionValue(textValue);
             case NUMERIC:
                 return textValue;
             case MONEY:
                 return parseMoney(textValue);
             default:
-                throw new IllegalArgumentException("Unsupported numeric type: " + columnType);
+                return textValue;
+        }
+    }
+
+    private Object parseNumericArray(PostgresDataType columnType, String textValue) {
+        switch (columnType) {
+            case INT2ARRAY:
```

Comment:
Not blocking, but we can probably have a mapping from array types to their corresponding scalar types in PostgresDataType class so that we can get rid of the switch block here and other handlers.